IdentityCore/src/workplacejoin/MSIDPkeyAuthHelper.m

```diff
@@ -106,11 +108,9 @@ + (nullable NSString *)createDeviceAuthResponse:(nonnull NSString*)authorization
             // Error should have been logged before this where there is more information on why the challenge data was bad
             MSID_LOG_INFO(context, @"PKeyAuth: Received PKeyAuth request with no challenge data.");
         }
-        
-        pKeyAuthHeader = [NSString stringWithFormat:@"PKeyAuth %@ Context=\"%@\", Version=\"%@\"", pKeyAuthHeader,[challengeData valueForKey:@"Context"],  [challengeData valueForKey:@"Version"]];
     }
     
-    return pKeyAuthHeader;
+    return [NSString stringWithFormat:@"PKeyAuth %@ Context=\"%@\", Version=\"%@\"", authToken, challengeContext, challengeVersion];
```

Comment:
For readability , I would still recommend capturing it as pKeyAuthHeader string and return that.

Reply:
Not sure if it improves readability much to be honest. I'd rather like to avoid seeing following pattern unless there's a good reason.